app/models/mapping_rule.rb

```diff
@@ -47,6 +67,14 @@ def as_json
     }
   end
 
+  def as_txt(padding: 0)
+    input_text = input.as_txt
+    output_text = output.as_txt
+    return if input_text.empty? || output_text.empty?
```

Comment:
Do we need to guard on empty strings or only nil? I suppose given this is calling `as_txt` I would expect _always strings_ and never nil (let's make sure that's true)

I thought we also decided we'd only output cases where there's a difference, right?

```suggestion
    input_text = input.as_txt
    output_text = output.as_txt
    
    return "" if input_text.blank? || output_text.blank?
    return "" if input_text == output_text
```

Reply:
we probably don't want to skip anything for shopify → google mappings because those are between 2 different systems/integrations

So we can perform `input_text == output_text` if source_integration_id == destination_integration_name

Reply:
Done.